usr/src/lib/libproc/aarch64/Pisadep.c

```diff
@@ -81,27 +81,57 @@ Pissyscall(struct ps_prochandle *P, uintptr_t addr)
 	if (Pread(P, &instr, sizeof (instr), addr) != sizeof (instr))
 		return (0);
 
-	if ((instr & 0xFE00001F) == 0xD4000001)
+	if ((instr & 0xFE00001F) == 0xD4000001) /* svc <imm16> */
 		return (1);
 
 	return (0);
 }
 
+int
+Pissyscall_indirect(struct ps_prochandle *P, uintptr_t addr)
+{
+	uint32_t instr;
+
+	if (Pread(P, &instr, sizeof (instr), addr) != sizeof (instr))
+		return (0);
+
+	return (instr == 0xd4000001);
```

Comment:
Where do we make the guarantee that svc#0 is indirect?

Reply:
In the system call table `sysent` system call 0 is traditionally the indirect syscall (on all platforms).  On platforms that implement indirect calls entirely in the C library (x86 and arm) the actual kernel-side entry point `indir()` just returns `nosys()`, and the implementation of `syscall` (and `syscall6`) in libc dictates the ABI for indirect calls.

In the ARM case this is that the syscall# is in `%x9`, the arguments in the ABI argument registers, and `svc #0` issued to enter the system.
on x86 this is that the syscall# is in `%eax` or `%rax`, the arguments in the ABI registers, and `sysenter` etc, used to enter the system.